core/src/main/java/org/lflang/diagram/synthesis/util/SynthesisErrorReporter.java

```diff
@@ -24,61 +24,71 @@
  ***************/
 package org.lflang.diagram.synthesis.util;
 
+import de.cau.cs.kieler.klighd.Klighd;
 import java.nio.file.Path;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
 import org.eclipse.emf.ecore.EObject;
 import org.lflang.ErrorReporter;
 
 /**
  * @author Alexander Schulz-Rosengarten
  */
 public class SynthesisErrorReporter implements ErrorReporter {
+
+  private boolean errorsOccurred = false;
+
   @Override
   public String reportError(String message) {
-    return null;
+    errorsOccurred = true;
+    Klighd.log(new Status(IStatus.ERROR, SynthesisErrorReporter.class, message));
+    return message;
   }
 
   @Override
   public String reportError(EObject object, String message) {
-    return null;
+    return reportError(message);
```

Comment:
I actually don't know. Probably @a-sr can provide more context. I added this forwarding of messages via klighd's logger so that we get proper error reporting in `lfd` and can also intercept errors when testing. I don't know what klighd's logger does when we use this from Eclipse or the LSP. The default behavior of the logger is to print to stdout. If this is the case also for Epoch and the LSP, then some additional output in the log should not be problematic. We can try this out, and if we see duplicated messages we can always register a status handler that just discards the messages.